app/users/services/ArchimatchUserService.py

```diff
@@ -72,70 +59,43 @@ def archimatch_user_create_password(cls, request):
         """
         try:
             data = request.data
-            request_keys = set(data.keys())
-            expected_keys = {
-                "email",
-                "password",
-                "confirm_password",
-            }
+            serializer = ArchimatchUserCreatePWSerializer(data=data)
+            serializer.is_valid(raise_exception=True)
+            validated_data = serializer.validated_data
+
+            email = validated_data.get("email")
+            password = validated_data.get("password")
+            confirm_password = validated_data.get("confirm_password")
 
-            # Validate input keys
-            cls.handle_user_data(request_keys, expected_keys)
-            req_email = data.get("email")
             # Check if user exists
-            if not ArchimatchUser.objects.filter(email=req_email).exists():
-                response_data = {
-                    "message": "User does not exist",
-                    "status_code": status.HTTP_404_NOT_FOUND,
-                }
-                return Response(
-                    response_data["message"],
-                    status=response_data["status_code"],
-                )
+            if not ArchimatchUser.objects.filter(email=email).exists():
+                raise ValidationError("User does not exist")
 
-            user = ArchimatchUser.objects.get(email=req_email)
-            password = data.get("password")
-            confirm_password = data.get("confirm_password")
+            user = ArchimatchUser.objects.get(email=email)
```

Comment:
user = ArchimatchUser.objects.get(email=email)
if not user:
                raise NotFound("User does not exist")

Reply:
already checked existence : 

```
# Check if user exists
            if not ArchimatchUser.objects.filter(email=email).exists():
                raise NotFound("User does not exist")

            user = ArchimatchUser.objects.get(email=email)
```